File: 백준/Bronze/11948. 과목선택/과목선택.c
#include <stdio.h>

int main(void) {
	int score[6] = { 0 }, sum = 0;
	for (int i = 0; i < 6; i++)
		scanf("%d", &score[i]);
	for (int i = 0; i < 4; i++) {
		for (int j = i + 1; j < 4; j++) {
			if (score[i] < score[j]) {
				int temp = score[i];
				score[i] = score[j];
				score[j] = temp;
			}
		}
	}
	if (score[5] > score[4]) sum += score[5];
	else sum += score[4];
	sum += score[0] + score[1] + score[2];
	printf("%d", sum);
	return 0;
}

File: 백준/Bronze/11948. 과목선택/과목선택.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int[] score = new int[4];
        for(int i = 0; i < 4; i++)
            score[i] = Integer.parseInt(br.readLine());
        int sum = 0;
        for(int i = 0; i < 2; i++) {
            int temp = Integer.parseInt(br.readLine());
            if(temp > sum) sum = temp;
        }
        Arrays.sort(score);
        for(int i = 1; i < 4; i++)
            sum += score[i];
        System.out.println(sum);
    }
}